weave/integrations/instructor/instructor_iterable_utils.py

```diff
@@ -0,0 +1,61 @@
+from functools import wraps
+from typing import Any, Callable, List, Optional
+
+from pydantic import BaseModel
+
+import weave
+from weave.trace.op_extensions.accumulator import add_accumulator
+
+
+def instructor_iterable_accumulator(
+    acc: Optional[BaseModel], value: BaseModel
+) -> List[BaseModel]:
+    if acc is None:
+        acc = [value]
+    if acc[-1] != value:
+        acc.append(value)
+    return acc
+
+
+def should_accumulate_iterable(inputs: dict) -> bool:
```

Comment:
The mypy error on L23 is because this dict is untyped.  If you know the shape, you might want to use `TypedDict`.  Otherwise you can assert that `inputs["stream"]` below is a bool before returning